lib/DXIL/DxilShaderFlags.cpp

```diff
@@ -778,17 +840,86 @@ ShaderFlags ShaderFlags::CollectShaderFlags(const Function *F,
   flag.SetSamplerDescriptorHeapIndexing(hasSamplerDescriptorHeapIndexing);
   flag.SetAtomicInt64OnHeapResource(hasAtomicInt64OnHeapResource);
   flag.SetAdvancedTextureOps(hasAdvancedTextureOps);
-  flag.SetWriteableMSAATextures(hasWriteableMSAATextures);
+  flag.SetWriteableMSAATextures(setWriteableMSAATexturesBasedOnGlobal
+                                    ? hasWriteableMSAATexturesGlobal
+                                    : hasWriteableMSAATextures);
   flag.SetWaveMMA(hasWaveMMA);
   // Only bother setting the flag when there are UAVs.
   flag.SetResMayNotAlias(canSetResMayNotAlias && hasUAVs &&
                          !M->GetResMayAlias());
   flag.SetSampleCmpGradientOrBias(hasSampleCmpGradientOrBias);
   flag.SetExtendedCommandInfo(hasExtendedCommandInfo);
+  flag.SetUsesDerivatives(hasDerivatives);
 
   return flag;
 }
 
 void ShaderFlags::CombineShaderFlags(const ShaderFlags &other) {
   SetShaderFlagsRaw(GetShaderFlagsRaw() | other.GetShaderFlagsRaw());
 }
+
+ShaderFlags
```

Comment:
As many times as I've cursed when I came to a called function without any description above it, I would like a description above this function. 
```suggestion
// Use the function properties `props` to determine the minimum shader model and
// flag requirements based on shader stage and feature usage.
// Compare that minimum required version to the values passed in with
// `minMajor` and `minMinor` and pass the maximum of those back through those
// same variables.
// Return a copy of the `ShaderFlags` object with any additional flags required
// by `props` set.
ShaderFlags
```
Note that if the returned value of this changes because of my comment below, the last line will have to change too.